exoscale_auth.py

```diff
@@ -29,3 +31,61 @@ def __call__(self, request):
             'Authorization': auth,
         })
         return request
+
+
+class ExoscaleV2Auth(AuthBase):
+    def __init__(self, key, secret):
+        self.key = key
+        self.secret = secret.encode('utf-8')
+
+    def __call__(self, request):
+        expiration_ts = int(time.time() + 10 * 60)
+        self._sign_request(request, expiration_ts)
+        return request
+
+    def _sign_request(self, request, expiration_ts):
+        auth_header = 'EXO2-HMAC-SHA256 credential={}'.format(self.key)
+        msg_parts = []
+
+        # Request method/URL path
+        msg_parts.append('{method} {path}'.format(
+            method=request.method, path=urlparse(request.url).path
+        ))
+
+        # Request body
+        msg_parts.append(body.encode('utf-8') if request.body else u'')
```

Comment:
This line has 2 issues: 

- first body does not exists, it should be request.body
- second, body.encode('utf-8') return bytes and not a string making the program cash at line 69.

Unfortunately fixing the 2 issues is not enough as while it allow the code to go the the end, the generated signature does not work.

(Test case: post query with data parameter).

Reply:
Indeed. With `request.body` being bytes already, `msg_parts` should be transformed into a list of bytes parts only.

Reply:
See #3